espnet2/enh/layers/dnn_iva.py

```diff
@@ -0,0 +1,140 @@
+"""DNN beamformer module."""
+from distutils.version import LooseVersion
+from typing import List
+from typing import Optional
+from typing import Tuple
+from typing import Union
+
+import logging
+import torch
+from torch.nn import functional as F
+from torch_complex.tensor import ComplexTensor
+
+from espnet2.enh.layers.iva import auxiva_iss
+from espnet2.enh.layers.complex_utils import stack
+from espnet2.enh.layers.complex_utils import to_double
+from espnet2.enh.layers.complex_utils import to_float
+from espnet2.enh.layers.mask_estimator import MaskEstimator
+
+
+is_torch_1_9_plus = LooseVersion(torch.__version__) >= LooseVersion("1.9.0")
+
+
+class DNN_IVA(torch.nn.Module):
+    """DNN mask based Beamformer.
+
+    Citation:
+        Multichannel End-to-end Speech Recognition; T. Ochiai et al., 2017;
+        http://proceedings.mlr.press/v70/ochiai17a/ochiai17a.pdf
```

Comment:
This docstring does not seem to correctly describe the actual algorithm.

Reply:
fixed